rules/opentitan.bzl

```diff
@@ -0,0 +1,203 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+# TODO(drewmacrae) this should be in rules_cc
+load("//rules:bugfix.bzl", "find_cc_toolchain")
+
+"""Rules to build OpenTitan for the RiscV target"""
+
+OPENTITAN_CPU = "@bazel_embedded//constraints/cpu:riscv32"
+OPENTITAN_PLATFORM = "@bazel_embedded//platforms:opentitan_rv32imc"
+
+_targets_compatible_with = {
+    OPENTITAN_PLATFORM: [OPENTITAN_CPU],
+}
+
+def _platforms_transition_impl(settings, attr):
+    return {"//command_line_option:platforms": attr.platform}
+
+_platforms_transition = transition(
+    implementation = _platforms_transition_impl,
+    inputs = [],
+    outputs = ["//command_line_option:platforms"],
+)
+
+def _elf_to_binary(ctx):
+    cc_toolchain = find_cc_toolchain(ctx)
+    outputs = []
+    for src in ctx.files.srcs:
+        binary = ctx.actions.declare_file("{}.{}".format(src.basename, ctx.attr.suffix))
+        outputs.append(binary)
+        ctx.actions.run(
+            outputs = [binary],
+            inputs = [src] + cc_toolchain.all_files.to_list(),
+            arguments = [
+                "--output-target",
+                ctx.attr.format,
+                src.path,
+                binary.path,
+            ],
+            executable = cc_toolchain.objcopy_executable,
+        )
+    return [DefaultInfo(files = depset(outputs), data_runfiles = ctx.runfiles(files = outputs))]
+
+elf_to_binary = rule(
```

Comment:
Honestly this isn't even ELF-specific. Not to bikeshed names, but I feel it wouldn't be *wrong* for this to be named `binary_to_image` (which is *really* what it's doing: it takes an ELF binary and relinks it as an .iso-like thing).

Given that we're redoing everything from scratch, this is a good time to re-evaluate our naming decisions.

Reply:
I'd suggest `bin_transform` or `obj_transform`, as `objcopy` can consume and emit many different kinds of binary object formats.

The other `elf_to_...` rules want to consume ELF files (IIRC).